lib/src/components/NxAccordion/__tests__/NxAccordion.test.tsx

```diff
@@ -5,313 +5,212 @@
-              <NxAccordion.Header className="header-class">
-                <span>Foo</span>
-              </NxAccordion.Header>
-            )
-          }),
-          header = component.find('summary');
-
-      expect(header).toHaveClassName('nx-accordion__header');
-      expect(header).toHaveClassName('header-class');
-      expect(getMountedComponent({ children: <NxAccordion.Header /> }).find('summary'))
-          .toHaveClassName('nx-accordion__header');
-    });
+    it('sets the provided className', function() {
+      const el = renderEl()!;
+      const customEl = renderEl({ className: 'foo' })!;
 
-    it('renders an icon with nx-accordion__chevron class as the first child of the header wrapper', function() {
-      const component = getMountedComponent({
-            children: (
-              <NxAccordion.Header>
-                <span>Foo</span>
-                <div className="nx-btn-bar">
-                  <NxButton />
-                </div>
-              </NxAccordion.Header>
-            )
-          }),
-          wrapper = component.find('summary').children();
-
-      expect(wrapper.children().first()).toMatchSelector(NxFontAwesomeIcon);
-      expect(wrapper.children().at(1)).toMatchSelector('span');
-      expect(wrapper.children().last()).toMatchSelector('div');
-
-      expect(wrapper.find(NxFontAwesomeIcon)).toHaveClassName('nx-accordion__chevron');
-    });
+      expect(customEl).toHaveClass('foo');
 
-    it('uses faChevronCircleDown as the icon when the accordion is closed', function() {
-      const component = getMountedComponent({
-            children: (
-              <NxAccordion.Header>
-                <span>Foo</span>
-              </NxAccordion.Header>
-            )
-          }),
-          header = component.find('summary');
-
-      expect(header.find(NxFontAwesomeIcon)).toHaveProp('icon', faChevronCircleDown);
-
-      const explicitOpenComponent = getMountedComponent({
-            open: false,
-            children: (
-              <NxAccordion.Header>
-                <span>Foo</span>
-              </NxAccordion.Header>
-            )
-          }),
-          explicitOpenHeader = explicitOpenComponent.find('summary');
-
-      expect(explicitOpenHeader.find(NxFontAwesomeIcon)).toHaveProp('icon', faChevronCircleDown);
+      for (const cls of Array.from(el.classList)) {
+        expect(customEl).toHaveClass(cls);
+      }
     });
 
-    it('uses faChevronCircleUp as the icon when the accordion is open', function() {
-      const component = getMountedComponent({
-            open: true,
-            children: (
-              <NxAccordion.Header>
-                <span>Foo</span>
-              </NxAccordion.Header>
-            )
-          }),
-          header = component.find('summary');
-
-      expect(header.find(NxFontAwesomeIcon)).toHaveProp('icon', faChevronCircleUp);
+    it('renders the NxAccordion.Header as a <summary>', function() {
+      const el = renderEl({ children:
```

Comment:
Don't forget you redefined `renderEl` in the Header `describe` block.  So by specifying an `NxAccordion.Header` in the `children` here, you are rendering a header nested inside another header.  Same in the next test below

Reply:
> Don't forget you redefined `renderEl` in the Header `describe` block. So by specifying an `NxAccordion.Header` in the `children` here, you are rendering a header nested inside another header. Same in the next test below

Updated the renderEl here: 0176eb8acf4d984deb4cb1a18659eb0341fe65b6